File: Practica4/RUBY/src/BetaPowerEfficientSpaceStation.rb

require_relative 'PowerEfficientSpaceStation'
require_relative 'BetaPowerEfficientSpaceStationToUI'

module Deepspace

class BetaPowerEfficientSpaceStation < PowerEfficientSpaceStation
	
	@@EXTRAEFFICIENCY = 1.2

    def initialize (s)
      super(s)
      d = Dice.new()
    end
    
    #override
    def fire
		
      if(d.extraEfficiency)
        retorno = super*@@EXTRAEFFICIENCY
      else
        retorno = super
      end
      return retorno
    end
    
    #Override
    def to_s
      return "(Beta) " + super
    end
	
	def getUIversion
      return BetaPowerEfficientSpaceStationToUI.new(self)
    end
    
end # class

end # module

File: Practica4/Marcianitos/src/deepspace/BetaPowerEfficientSpaceStation.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;

/**
 *
 * @author mario
 */
public class BetaPowerEfficientSpaceStation extends PowerEfficientSpaceStation{
    
    private static final float EFFIENCYFACTOR=1.2f;
    private Dice dice;
    
    BetaPowerEfficientSpaceStation(SpaceStation station){
        super(station);
        dice=new Dice();
    }
    
    @Override
    public float fire(){
        
        float retorno;
        
        if(dice.extraEfficiency()){
            
            retorno=EFFIENCYFACTOR*super.fire();
        }
        else{
            retorno=super.fire();
        }
        
        return retorno;
    }
	
	
	@Override
	public BetaPowerEfficientSpaceStationToUI getUIversion (){
		return (new BetaPowerEfficientSpaceStationToUI(this));
	}
    
}
